scvi/autotune/_metrics.py

```diff
@@ -0,0 +1,86 @@
+from typing import Optional
+
+import numpy as np
+import pandas as pd
+from sklearn.metrics import silhouette_samples, silhouette_score
+
+from scvi.model.base import BaseModelClass
+
+
+def silhouette_metric_labels_batch(
+    model: BaseModelClass,
+    labels_key: str,
+    batch_key: str,
+    sample_size: int = 1000,
+    unknown_label: Optional[str] = None,
+) -> float:
+    """
+    Batch- and label-wise silhouette.
+    Parameters
+    ----------
+    model
+        A scvi model.
+    labels_key
+        The key of the labels.
+    batch_key
+        The key of the batch.
+    sample_size
+        Sample size for silhouette. Randomly subsets the data
+    Returns
+    -------
+    Sum of asw for batch and for labels. Scores are scaled such that
+    2 is the best score and 0 is the worst.
+    Notes
+    -----
+    This function is influence by the following code:
+    https://github.com/theislab/scib/blob/main/scib/metrics/silhouette.py
+    """
+    model.is_trained_ = True
+    latent = model.get_latent_representation()
+    model.is_trained_ = False
```

Comment:
What is the reason for switching this off?

Reply:
Oh I see, could you add a note for the reason behind it?

Reply:
Included the reasoning as a comment, were you referring to a note in the docstring instead?